Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+            // Providing this table enables the retrieval of a location's floor level ID.
+            // positioningID - an ID which identifies a specific row in the positioningTable that should be used for setting up IPS.
+            return IndoorsLocationDataSource(
+                positioningTable: positioningTable,
+                pathwaysTable: pathwaysLayer.featureTable as? ArcGISFeatureTable,
+                levelsTable: levelsLayer.featureTable as? ArcGISFeatureTable,
+                positioningID: positioningID
+            )
+        }
+        
+        /// The method that updates the location when the indoors location datasource is triggered.
+        func updateDisplayOnLocationChange() async throws {
+            for await location in locationDisplay.dataSource.locations {
+                // Since this listens for new location changes, it is important
+                // to ensure any blocking UI is dismissed once location updates begins.
+                // Floors in location are zero indexed however floorManager levels begin at one. Since
+                // it is necessary to display the same information to the user as the floor manager levelNumber
+                // one is added to the floor level value.
+                if let floorLevel = location.additionalSourceProperties[.floor] as? Int,
+                   (floorLevel + 1) != currentFloor {
+                    // Sets the currentFloor to the new floor level and adds one, since location uses
+                    // zero based flooring system.
+                    currentFloor = floorLevel + 1
+                    // The floor manager is used to filter so that only the data from the current floor is
+                    // displayed to the user.
+                    if let floorManager = map.floorManager {
+                        floorManager.levels.forEach {
+                            $0.isVisible = currentFloor == $0.levelNumber
+                        }
+                    }
+                }
+                // This indicates whether the location data was sourced from GNSS (Satellites), BLE (Bluetooth Low Energy)
+                // or AppleIPS (Apple's proprietary location system.
+                let source = location.additionalSourceProperties[.positionSource] as? String ?? ""
+                switch source {
+                case "GNSS":
+                    satelliteCount = location.additionalSourceProperties[.satelliteCount] as? Int ?? 0
+                default:
+                    sensorCount = location.additionalSourceProperties[.transmitterCount] as? Int ?? 0
+                }
+                horizontalAccuracy = location.horizontalAccuracy
+                // Updates every time the location changes.
+                getStatusLabelText(source: source)
+            }
+        }
+        
+        /// Updates the labels on the view with the current state of the indoors data source.
+        private func getStatusLabelText(source: String) {
```

Comment:
The prefix `get` implies that the method returns something, but in fact it doesn't return anything. Not only that, this is something the view should be doing, not the model. The model shouldn't be determining user-facing strings, particularly ones containing values that should be formatted appropriately. (After all, we want to model best practices to our customers, which includes formatting data appropriately.) The view should be doing that. Please move this logic and the corresponding properties into the view. The view should use the state of the model to determine what should be displayed.